src/core/config/border_routing.h

```diff
@@ -88,6 +88,28 @@
 #define OPENTHREAD_CONFIG_BORDER_ROUTING_TRACK_PEER_BR_INFO_ENABLE OPENTHREAD_CONFIG_BORDER_ROUTING_USE_HEAP_ENABLE
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_BORDER_ROUTING_REACHABILITY_CHECK_ICMP6_ERROR_ENABLE
+ *
+ * Define to 1 to allow Routing Manager to check for reachability of messages being forwarded by the BR and determine
+ * whether to send an ICMPv6 Destination Unreachable error back to the sender.
+ *
+ * Specifically, if the Border Router (BR) decides to forward an IPv6 message outside the AIL and the message's
+ * source address matches a BR-generated ULA OMR prefix (with low preference), and the destination is unreachable
+ * using this source address, then an ICMPv6 Destination Unreachable message is sent back to the sender.
+ *
+ * For example, this situation can occur when a local, non-infrastructure-derived ULA OMR prefix is published alongside
+ * a `::/0` route (due to discovered PIO/RIO prefixes by the BR). A Thread mesh device may try to reach addresses
+ * beyond the local AIL (e.g., the global internet) using the ULA OMR prefix, which would be unreachable.
```

Comment:
"using ... prefix"  should be "using ... address" here -> prefix isn't used in the packet.

```suggestion
 * beyond the local AIL (e.g., the global internet) using its ULA OMR address as source, which would be unreachable.
```